src/controller/CHIPDeviceController.h

```diff
@@ -581,6 +664,9 @@ class DLL_EXPORT DeviceCommissioner : public DeviceController, public SessionEst
     Callback::Callback<DefaultFailureCallback> mOnRootCertFailureCallback;
 
     PASESession mPairingSession;
+
+    static PingParams mPingParams;
```

Comment:
Is this a spec requirement? 

This feels very out of place,
Why are pingparams static? why dshould every controller support pingparams and echo clients?

Reply:
mPingParams is used by OnHandleEchoResponse which need to be static as the callback of EchoClient.

Reply:
Please split out echo handling more - I believe "commisioner" should not be directly responsible for echo/ping parameters and callbacks. If some integration is required, please make it minimal. OnHandle.... should not be part of the commisioner.

Reply:
Acked, refactored the Callback handler in Echo with delegate pattern, we don't need to use the static member functions and variables for handling the callback.